presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -460,7 +460,7 @@ public JdbcOutputTableHandle beginInsertTable(ConnectorSession session, JdbcTabl
             ImmutableList.Builder<String> columnNames = ImmutableList.builder();
             ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
             ImmutableList.Builder<JdbcTypeHandle> jdbcColumnTypes = ImmutableList.builder();
-            for (JdbcColumnHandle column : getColumns(session, tableHandle)) {
```

Comment:
The nice thing about it is that  you are optimizing single new JDBC connection that would `getColumns` do.